proposals/20201026-creating-control-plane.md

```diff
@@ -0,0 +1,322 @@
+* **Super Master** - The control plane of the super cluster.
+* **NestedControlPlane(NCP)** - The control plane that is hosted on the super cluster. 
+* **NestedControlPlaneTemplate(NCPT)** - NCPT contains all information required to create the NCP.
+* **NestedPKI(NPKI)** - NPKI includes TLS certificates and kubeconfigs required by components of the NCP.
+* **NestedEtcd(NEtcd)** - The etcd that belongs to the control plane of the nested cluster.
+* **NestedKAS(NKAS)** - The kube-apiserver which belongs to the control plane of the nested cluster.
+* **NestedKCM(NKCM)** - The kube-control-manager which belongs to the control plane of the nested cluster.
+
+## Summary
+
+The goal of this proposal is to define a basic/native way of creating control plane 
+components~(kube-apiserver~(KAS), Etcd, kube-controller-manager~(KCM)) for the CAPN.
+
+## Motivation
+
+CAPN aims at providing control-plane level isolation while sharing physical resources 
+among control planes. There exist various ways of creating isolated control-planes. 
+For example, one can create NCPs through cloud providers' Kubernetes services or has 
+a separate cluster, just running NCPs. However, the most straightforward way is directly 
+running the nested control-plane components on the underlying clusters. In this proposal, 
+we present a way of hosting the nested control-plane on the underlying clusters. By 
+using the proposed mechanism, the user can set up the CAPN environment on any standard 
+Kubernetes cluster without using any third-party platforms or resources.
+
+### Goals
+
+- Define the CRD of each nested component.
+- Define how NEtcd, NKAS and NKCM controllers cooperate to create components of NCP.
+
+### Non-Goals
+
+- Define how NCP controller works.
+- Define how NPKI controller works.
+- Define a mechanism/interface that is portable to other existing controllers, e.g., etcd-operator.
+
+## Proposal
+
+### Bootstrap
+
+The NCP controller will need to create some resources before sub-controllers 
+start creating the real control plane components. 
+
+* **ClusterNamespace** is used to store all resources for the NCP.
+
+* **NKAS service**  is used to visit the NKAS. Since IP/domain 
+name is required when generating TLS certificates and kubeconfigs of NKAS, this 
+service needs to be created in advance.
+
+* **NestedCRs** includes NPKI, NEtcd, NKAS, and NKCM. The NCP controller needs to 
+create all CRs for sub-controllers. Among them, NEtcd, NKAS, and NKCM will be 
+created based on the NCPT referred by the NCP.
+
+After creating the above prerequisites, sub-controllers will start creating the workload of each component.
+
+### Creation
+
+By default, we will use the StatefulSet to create each component. The readiness 
+and liveness probe will be used, and we will mark each component as ready only 
+when the corresponding Statefuleset is ready. As the KAS cannot work without 
+available Etcd and the KCM cannot run without KAS, the three components need to 
```

Comment:
If they are created all at once and have liveness/readiness could they be eventually available?